### PR TITLE
Fix decompose of AvgPool2D to account for named attributes and skip runtime reinterpreting shapes

### PR DESCRIPTION
**Pull request by dgolubovicTT**
Sloves #560, solves #559. 

This PR includes:
    - AvgPool2D fix decompose to account for named attributes that are needed for mlir lowering.
    - skip lower_reinterpret_shape in optimization passes to avoid changing expected shapes on input.
    - Test  test_avg_pool2d to test this.

### CODE REVIEW

File: forge/csrc/forge_passes.cpp [hunk 1]
```diff
@@ -157,7 +157,8 @@ void run_optimization_graph_passes(graphlib::Graph *graph)
 
     passes::hoist_transforms_to_inputs(graph);
     passes::erase_consecutive_reshape(graph, true);
-    passes::lower_reinterpret_shape(graph);
+    // skip lower_reinterpret_shape in optimization passes to avoid changing expected shapes on inputs
```
Comment (nvukobratTT) [inline forge/csrc/forge_passes.cpp:160]: Do you see a value in fully removing this code section? If we need a reference in future, we can always check PyBuda repo.

File: forge/forge/op/eval/forge/pooling.py [hunk 2]
```diff
@@ -478,15 +478,20 @@ def decompose(type, attr, dc, inputs):
         # If global average
         if y == kH and x == kW and ((stride[0] == kH and stride[1] == kW) or all(pad == 0 for pad in padding)):
             if channel_last:
-                result = dc.op("reshape", [activations], (w, 1, y * x, cin))
-                result = dc.op("reduce_avg", [result], (-2,))
-                result = dc.op("reshape", [result], (w, 1, 1, cin))
+                # for reshape use dc.op_with_named_attrs
```
Comment (nvukobratTT) [inline forge/forge/op/eval/forge/pooling.py:481]: Do we need this comment? In the future, we should transfer to using named attributes for each op :))

**Review by nvukobratTT — APPROVED** [review]

File: forge/forge/op/eval/forge/pooling.py (hunk 2 above)
Comment (dgolubovicTT) [reply forge/forge/op/eval/forge/pooling.py:481]: Yeah it is unnecessary...

File: forge/csrc/forge_passes.cpp (hunk 1 above)
Comment (pilkicTT) [reply forge/csrc/forge_passes.cpp:160]: +1 just remove it 😄

i would also remove that pass entirely.

Comment (dgolubovicTT) [reply forge/csrc/forge_passes.cpp:160]: Hmmm, I wanted to leave some clue why we are skipping this for now. Looking back to PyBuda doesn't sound amazing to me. Do we have some third option?

Comment (nvukobratTT) [reply forge/csrc/forge_passes.cpp:160]: > Do we have some third option?

To remove it and never look back? 😂

**Pull request merged**